parties/opkclient.go

```diff
@@ -23,49 +25,68 @@ type MFACos interface {
 }
 
 type OpkClient struct {
-	PktCom      []byte
-	Signer      *pktoken.Signer
-	Op          OpenIdProvider
-	MFACosigner MFACos
+	Pkt           *pktoken.PKToken
+	SigningKey    crypto.Signer
+	UserPublicKey jwk.Key // Requires "alg" header to be set
+	Gq            bool
+	Op            OpenIdProvider
+	MFACosigner   MFACos
 }
 
 func (o *OpkClient) OidcAuth() ([]byte, error) {
-	nonce, err := o.Signer.GetNonce()
+	// Use provided public key to generate client instance claims
+	cic, err := clientinstance.NewClaims(o.UserPublicKey, map[string]any{})
```

Comment:
The extra CIC claims here need to come from somewhere, could be a parameter to this function or a field on the `OpkClient` struct.